server/src/main/java/io/druid/segment/realtime/plumber/RealtimePlumber.java

```diff
@@ -417,7 +459,7 @@ public void doRun()
               synchronized (hydrant) {
                 if (!hydrant.hasSwapped()) {
                   log.info("Hydrant[%s] hasn't swapped yet, swapping. Sink[%s]", hydrant, sink);
-                  final int rowCount = persistHydrant(hydrant, schema, interval);
+                  final int rowCount = persistHydrant(hydrant, schema, interval, null);
```

Comment:
yes, earlier/latest depending upon what behavior is set in the firehose it will either duplicate or lose data. again, this one is hard to solve in existing code structure as there is not way to get handle to commit metadata at this point.